File: src/api/endpoints/websocket/WebsocketEndpoint.ts
import { HubConnection } from "@microsoft/signalr";
import { Logger } from "ts-log";

import { HeaderProvider } from "../../../Func";
import ApiEndpoint from "../ApiEndpoint";
import TokenManager from "../auth/TokenManager";
import DefaultApiConfig from "../DefaultApiConfig";
import WebsocketManager from ".//WebsocketManager";

/**
 * Endpoints for interacting with emergencies.
 * */
export default class WebsocketEndpoint extends ApiEndpoint {
  constructor(config: DefaultApiConfig, tokenManager: TokenManager, log?: Logger, headerProvider?: HeaderProvider) {
    super(config, tokenManager, log, headerProvider);
  }

  private websocketManager = new WebsocketManager(this.config, this.tokenManager, this.log);

  protected override endpoint(): string {
    return "websocket";
  }

  /**
   * Gets realtime updates.
   *
   * */
  public async initialize(): Promise<HubConnection> {
    if (this.websocketManager === undefined) {
      throw new Error("WebsocketManager is undefined");
    }
    return await this.websocketManager.establishConnection();
  }
}

File: src-dotnet/api/endpoints/websocket/WebsocketEndpoint.cs
namespace MedrunnerApiClient.Api.Endpoints.Websocket;

    /// <summary>
    /// Endpoints for interacting with websocket/realtime updates.
    /// </summary>
    public class WebsocketEndpoint : ApiEndpoint
    {
        private readonly WebsocketManager _websocketManager;

        public WebsocketEndpoint(ApiConfig config, TokenManager tokenManager, ILogger logger)
            : base(config, tokenManager, logger)
        {
            _websocketManager = new WebsocketManager(config, tokenManager, logger);
        }

        protected override string Endpoint => "websocket";

        /// <summary>
        /// Gets realtime updates (establishes websocket connection).
        /// </summary>
        public async Task<object> InitializeAsync()
        {
            // Replace object with actual connection type if available
            return await _websocketManager.EstablishConnectionAsync();
        }
    }
    